eip-draft-smartcontract-fee-share: Transaction fee share by smart contract deployers (#3279)
ethereum/EIPs: EIPS/eip-draft-smartcontract-fee-share.md

---

> eip: 3279

Reviewer: File name should be `eip-3279.md`.

> eip: <to be assigned>

Reviewer 2: ```suggestion
eip: 3279
```

title: Smart Contract Deployer Transactions Fee Share

> author: @tarekskr

Reviewer 2: ```suggestion
author: tarekskr (@tarekskr)
```
Handle should be in parenthesis. Please use the name to the left as your full name or moniker you'd like to officially be referred to as.

> status: Draft

Reviewer: ```suggestion
discussions-to: <URL>
status: Draft
```
`discussions-to` is required.  Recommend creating a thread on https://ethereum-magicians.org/ and putting it here.

type: Standards Track
category: Core
created: 2021-02-20
---

## Simple Summary

> This EIP proposes that a smart contract deployer should get a share of all transaction fees generated by the smart contract.

Reviewer: EIP's don't propose, they assert:
```suggestion
Makes the deployer of a contract receive a percentage of Ethereum transaction fees.
```

## Motivation
A lot of smart contract developers are not able to capture any value from their widely used smart contracts, and are thus forced to deploy meaningless tokens in order to be able to do so. If developers get a small share of all the fees generated by users interacting with their smart contracts, it will potentially solve the aforementioned problem, in addition to providing huge incentives for developers to choose Ethereum over other platforms for their applications.

> 

Reviewer 2: Missing specification, rationale, and security considerations sections.

Author: > Missing specification, rationale, and security considerations sections.
Would really love any help I could get with this 🙏

Reviewer: Every EIP needs a technical champion.  Keep in mind, EIPs are meant to be technical standards, not governance discussions.  If you just want to get the discussion going about this idea I recommend starting a thread over at https://ethresear.ch/ or https://ethereum-magicians.org/ rather than starting with an EIP.

Author: @MicahZoltu oh I see, thank you so much for the clarification. I will start a thread on https://ethereum-magicians.org/. 

I'd also be really grateful if you have any recommendations on who would be interested to be a technical champion for this.
